### ISSUE TITLE
T - Subscription: 'Become a monthly supporter' embedded on FAQ page

### ISSUE THREAD

**Issue by tobinmori** [mdn/kuma#6578]
**Description**:
Logged in users who click on subscribe in the banner see embedded widget on FAQ page to become monthly subscribers

**Acceptance Criteria**:
- [x] when logged in users click on subscribe in the banner they see the FAQ page to become monthly subscribers
- [x] has checkbox to verify that user wants to become a subscriber and acts as a gate to Stripe modal (no data needs to be stored upon confirmation)
- [x] Text in widget is identical to design for legal reasons (text in task issue)
- [x] If user clicks subscribe without having checked the box, an error message is shown
- [x] If the checkbox is checked and the user clicks on subscribe, a stripe modal is shown

**Design**:
https://app.abstract.com/projects/9decc480-70fc-11e9-9453-4f3fbffea7ce/branches/64547606-339e-494d-8d45-559aa506d80a/commits/6ae7cc2467cf36a609b9b2bc5189dabf9c5c4e7e/files/96E76C7B-2AF8-4473-A39E-E81890F91955/layers/E3B8DB66-C6A4-4BDC-A2FF-A80A02DAC2E8?collectionId=0750a16f-5979-4db7-9b1b-d4d676607dcd&collectionLayerId=2536d7fc-cd8e-4b9b-9580-615dc1f76e70&mode=design

**Notes**:

**Comment by tobinmori** [comment]
@Gregoor this issue can began after @schalkneethling finishes the 'bare bones' version of #6577

**Comment by schalkneethling** [comment]
> @Gregoor this issue can began after @schalkneethling finishes the 'bare bones' version of #6577

Also, just a heads up @Gregoor. This may be deferred to the next sprint as we will not be adding this interaction in this sprint. Instead, we will merely send users to the edit profile page where they will complete the sign up using the feature you have already implemented there.

**Comment by Gregoor** [comment]
Good to know, thanks! I'll hold my horses then 🐴

**Comment by tobinmori** [comment]
@Gregoor I've moved this one out of sprint Bravo for now. We can move it in once the bare bones banner is in.

**Comment by tobinmori** [comment]
We decided to eliminate this step, so closing this.

**Comment by schalkneethling** [comment]
![Screenshot 2020-03-23 at 12 25 04](https://user-images.githubusercontent.com/10350960/77307099-5663d700-6d01-11ea-9cf6-5bb21c307646.png)

Confirming that the above is the design and wording that is being implemented.

Also, the current implementation on the edit profile page sets the users email address in a hidden field like so:

```
<input type="hidden" name="stripe_email" value="{{ edit_user.email }}">
```

For this implementation there are two things to consider:

1. Not in this sprint but eventually, a user visiting the page might not be signed in and so, we do not have their email address to set
2. The data provided by the user provider in React does not contain the currently logged in user's email address so, we do not even have this data for logged in users

The questions then are:
1. Do we just leave this off for the new implementation? The Stripe modal does provide the ability for the user to enter their email address
2. Do we add an email address input field for all users
3. Do we leave the email property as empty for non-authenticated users, and then expose a logged-in user's email address via WhoAMI so it is available to the user provider? Are there any privacy or security concerns around exposing that information in such a way?

@tobinmori @atopal @malqinneh Thoughts?

**Comment by schalkneethling** [comment]
Oh and another thing. Currently, the payments landing page is public so, once this issue lands, so will the subscription component.

I am guessing that is not what we want. It is not mentioned here but I am assuming we do not want to make the entire page not public so, we should put the component behind a waffle flag.

That is one piece but, do we also want to implement the logic to only show it to authenticated users?

At this point, I am kinda leaning towards having Waffle control all of this and not implementing an additional conditional gate.

@tobinmori @atopal Thoughts?

**Comment by peterbe** [comment]
CC @Gregoor who wrote the Jinja version. 
Putting this on the Subscription landing page will require a considerable refactoring. Perhaps this issue should be assigned to him instead?

I don't remember why the Jinja based solution would form-post in the email but I think it's because you can, in the Stripe modal, override your preferred email for Stripe. 

Regarding waffling, why can't we just use the same waffle flag (`subscription`) that's currently used on the Edit Profile page? 
The page loads like this from SSR + CDN:
<img width="1316" alt="Screen Shot 2020-03-23 at 6 53 58 AM" src="https://user-images.githubusercontent.com/26739/77309720-872d1780-6cd3-11ea-9a64-106eed9a20a7.png">
Then, after the `/api/v1/whoami` has finished loading we'll know if `waffle.flags.subscription` is truthy and client-side React code in https://github.com/mdn/kuma/blob/master/kuma/javascript/src/payments/index.jsx re-renders but this time with the Stripe stuff enabled. 

The thing that I don't get is; what should it look like if you're NOT authenticated. I.e. `waffle.flags.subscription` is true but `is_authenticated` is false.

**Comment by Gregoor** [comment]
Yup, you remembered correctly Peter. Stripe needs to associate a customer with an email address, and users need to be able to see which address that is (which is why we need to supply it to the stripe modal).
Happy to take it on or wire it up once the visuals/flow is done.

**Comment by schalkneethling** [comment]
> Yup, you remembered correctly Peter. Stripe needs to associate a customer with an email address, and users need to be able to see which address that is (which is why we need to supply it to the stripe modal).
> Happy to take it on or wire it up once the visuals/flow is done.

@Gregoor, let me open the PR with what I have and we can chat from there. Sounds good?

**Comment by schalkneethling** [comment]
BTW, if you pass an empty value for the `email` property, the Stripe modal has a field to enter it.

**Comment by Gregoor** [comment]
@schalkneethling sounds good :+1:

Wrt stripe modal email, I remember Kadir asking for the email to be automatically filled-in. Unfortunately that answer apparently wasn't given in a GitHub issue (or I'm not patient enough to find the right now), so I can't link to it here.

**Comment by schalkneethling** [comment]
> @schalkneethling sounds good 👍
> 
> Wrt stripe modal email, I remember Kadir asking for the email to be automatically filled-in. Unfortunately that answer apparently wasn't given in a GitHub issue (or I'm not patient enough to find the right now), so I can't link to it here.

Oh it makes sense for it to be there but, the user provider does not provide it ;) I guess it is not exposed via WhoAMI

**Comment by tobinmori** [comment]
Thanks for raising this @schalkneethling 

> The thing that I don't get is; what should it look like if you're NOT authenticated. I.e. `waffle.flags.subscription` is true but `is_authenticated` is false.

My first instinct would be to simply not show it in this case, since the user is NOT authenticated. 

OR. . .have a design that keeps the widget and says something to the effect of 'Log in here to contribute!'

@atopal @malqinneh what do you think?

**Comment by Gregoor** [comment]
> 
> 
> > @schalkneethling sounds good 👍
> > Wrt stripe modal email, I remember Kadir asking for the email to be automatically filled-in. Unfortunately that answer apparently wasn't given in a GitHub issue (or I'm not patient enough to find the right now), so I can't link to it here.
> 
> Oh it makes sense for it to be there but, the user provider does not provide it ;) I guess it is not exposed via WhoAMI

happy to add that to your PR!

**Comment by tobinmori** [comment]
Note: the conclusion here is that we won't display the page unless the user is authenticated.

**Comment by schalkneethling** [comment]
> Note: the conclusion here is that we won't display the page unless the user is authenticated.

Hey @tobinmori, I might not have been clear. We will show the page, we will just not show the subscription form component.

**Comment by malqinneh** [comment]
> > Note: the conclusion here is that we won't display the page unless the user is authenticated.
> 
> Hey @tobinmori, I might not have been clear. We will show the page, we will just not show the subscription form component.

Hey Schalk, we do have mock-ups for non logged in users *and* non registered users: https://github.com/mdn/kuma/issues/6556
